internal/repo/order.go

```diff
@@ -0,0 +1,158 @@
+package repo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/jackc/pgx/v5"
+	"gophermart/internal/entity"
+	"gophermart/pkg/postgres"
+	"time"
+)
+
+type OrderRepo struct {
+	*postgres.Postgres
+}
+
+func NewOrder(pg *postgres.Postgres) *OrderRepo {
+	return &OrderRepo{pg}
+}
+
+func (orr *OrderRepo) Get(ctx context.Context, number entity.OrderNumber) (*entity.Order, error) {
+	query := `select user_id, order_nr, status, created_at from "order" where order_nr = $1`
+	o := &entity.Order{}
+	err := orr.Pool.QueryRow(ctx, query, number).Scan(&o.UserID, &o.Number, &o.Status, &o.UploadedAt)
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return nil, entity.ErrOrderNotFound
+		}
+		return nil, fmt.Errorf("OrderRepo - Get - orr.Pool.QueryRow: %w", err)
+	}
+	return o, nil
+}
+
+func (orr *OrderRepo) Add(ctx context.Context, order entity.Order) error {
```

Comment:
```suggestion
func (orr *OrderRepo) Add(ctx context.Context, tr *pgx.Tx order entity.Order) error {
```
вот так можно прокинуть транзакцию внутрь